### PR TITLE
apply miss_hit linter

### PR DESCRIPTION
**Pull request by Remi-Gau**
@marcobarilari you were asking for some styling tool that would help you know where to put spaces and so on.

This PR is the results of just typing this in the command line. Many of things this linter is not happy with are automatically fixed. Others throw a warning.

```
mh_style.py . --fix
```

From this project: https://florianschanda.github.io/miss_hit/style_checker.html

The main difference vizually is that it indents the functions by default, but the matlab editor can do that too if you set it up that way.

I am tempted to start adding this linter to our matlab projects so we don't have to **think** about formating.

@CerenB @marcobarilari 
What do you think?

### CODE REVIEW

**Comment by marcobarilari** [comment]
Can I apply it to real life? From a quick view, it looks fantastic. I'll go deeper later but first question a stupid question, can I ask it what changes I would like to do/don't apply?  

I see it is placing the `;` no matter what, in some cases might be useless and matlab would complain

**Comment by Remi-Gau** [comment]
> Can I apply it to real life? From a quick view, it looks fantastic. I'll go deeper later but first question a stupid question, can I ask it what changes I would like to do/don't apply?

It will only fix things automatically if you pass the `--fix` argument.

> I see it is placing the `;` no matter what, in some cases might be useless and matlab would complain

You can tweak the rules by creating a `miss_hit.cfg` config file.

The one i have created contains this to limit the amount of warnings and also capture our camelCase rule for the function names. It cannot check variable names (yet?).

```
line_length: 100
regex_function_name: "[a-z]+(([A-Z]|[0-9]){1}[a-z]+)*"
suppress_rule: "copyright_notice"
```

if you want to prevent the auto-fix of semi-colons then add:
`suppress_rule: end_of_statements`

**Comment by Remi-Gau** [comment]
Unless I hear screams of outrage I will make Miss_Hit the official linter tool for this repo in 24 hours.

Will try to see if I can add to the Travis continuous integration so we get it checked automatically when we open a PR and push.

File: CPP_getResponseDemo.m [hunk 1]
```diff
@@ -32,41 +33,35 @@
...
 KbName('UnifyKeyNames');
 
-
 % we ask PTB to tell us which keyboard devices are connected to the computer
 [cfg.keyboardNumbers, cfg.keyboardNames] = GetKeyboardIndices;
 
-cfg.keyboardNumbers
-cfg.keyboardNames
-
+cfg.keyboardNumbers;
```
Comment (marcobarilari) [inline CPP_getResponseDemo.m:46]: here I think we want to see these two printed in the command window, right? Then we could use:

```suggestion
fprintf('\n i% \n', cfg.keyboardNumbers);
```

File: CPP_getResponseDemo.m [hunk 2]
```diff
@@ -32,41 +33,35 @@
...
+cfg.keyboardNames;
```
Comment (marcobarilari) [inline CPP_getResponseDemo.m:47]: here as well

```suggestion
fprintf('\n s% \n, cfg.keyboardNames);
```

File: pressSpace4me.m [hunk 3]
```diff
@@ -1,19 +1,17 @@
 function pressSpace4me
```
Comment (marcobarilari) [inline pressSpace4me.m:1]:
```suggestion
function pressSpace4Me
```

Comment (marcobarilari) [reply pressSpace4me.m:1]: actually should be `pressSpaceForMe` in line with `degToPix`, though I like the numbers feel so yo

File: printCredits.m [hunk 4]
```diff
@@ -1,42 +1,41 @@
...
 
-version = '0.0.1';
+    version = '0.0.1';
 
-contributors = {...
-    'Rémi Gau', ...
-    'Marco Barilari'};
+    contributors = { ...
+        'Rémi Gau', ...
+        'Marco Barilari'};
```
Comment (marcobarilari) [inline printCredits.m:7]:
```suggestion
        'Marco Barilari', ...
        'Ceren Battal'};
```

File: testKeyboards.m [hunk 5]
```diff
@@ -22,44 +28,40 @@ function testKeyboards(cfg)
...
-text2 = '\nThese are the devices currently connected.\n\n';
+    switch keyboardType
+        case 1
+            keyboardType = 'experimenter keyboard';
+        case 2
+            keyboardType = 'response box';
+    end
 
-errorText = 'No key was pressed. Did you configure the keyboards properly? See message above for more info.';
+    text1 = ['\nYou asked for this keyboard device number to be used as ' keyboardType '.\n\n'];
```
Comment (marcobarilari) [inline testKeyboards.m:40]:
```suggestion
    text1 = '\nYou asked for this keyboard device number to be used as ' keyboardType '.\n\n';
```

File: pressSpace4me.m (hunk 3 above)
Comment (Remi-Gau) [reply pressSpace4me.m:1]: Good point.
Will make a change to both.

Comment (marcobarilari) [reply pressSpace4me.m:1]: If you find a way to change a var name across files in a path do let me know!

**Pull request merged**